### PR TITLE
[11.0][IMP] purchase_date_planned_manual: Avoid merging lines with different date_planned

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Solves https://github.com/OCA/purchase-workflow/pull/422#issuecomment-311287468.

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
Please, could you check @lreficent ?

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
It could work, but please can you add a test case to be sure?

**Comment by MiquelRForgeFlow** [comment]
Tests added :green_apple:

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Tested locally, it is not merging prcourements into the same line but planned date is not respected in PO
image

**Comment by MiquelRForgeFlow** [comment]
Fixed

**Comment by MiquelRForgeFlow** [comment]
ping: @lreficent

File: purchase_date_planned_manual/models/purchase_order.py [hunk 1]
```diff
@@ -52,3 +52,24 @@ def action_delayed_line(self):
...
             'arrive late.') % Dt.to_string(Dt.context_timestamp(
                 self, Dt.from_string(self.date_planned))))
+
+    def _merge_in_existing_line(self, product_id, product_qty, product_uom,
+                                location_id, name, origin, values):
+        if self.date_planned == values.get('date_planned'):
+            return super(PurchaseOrderLine, self)._merge_in_existing_line(
+                product_id, product_qty, product_uom,
+                location_id, name, origin, values)
+        else:
```
Comment (nikul-serpentcs) [inline purchase_date_planned_manual/models/purchase_order.py]: @mreficent here `else` is needed ??

I think It Should be a better :smiley: 
```
if self.date_planned == values.get('date_planned'):
    return super(PurchaseOrderLine, self)._merge_in_existing_line(
        product_id, product_qty, product_uom,
        location_id, name, origin, values)
return False
```

Comment (MiquelRForgeFlow) [reply purchase_date_planned_manual/models/purchase_order.py]: done

**Review by HviorForgeFlow — APPROVED** [review]

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@mreficent can you forward port to 12.0?

**Pull request merged**